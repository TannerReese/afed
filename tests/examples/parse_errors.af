# Test parsing errors

# Redefinition error
{ x: 4, x: 5},

# Circular dependence error
{
    foo: {x: 5 * bar.y = ``},
    bar: {y: baz.z - 2 = ``},
    baz: {z: 7 / foo.x = ``},
},

# Extra content in map or array
{1, "zing zing", 4n  kd ( $ $, "this is": "correct"} = ``,
{ "good", second: njd %% & 62 ] , third: "good"} = ``,
["this", "next"  ( % ^&, "last"] = ``,

# Redefinition with internal definition
{
    x: 4 = ``,
    4 * (x: 3) = ``,
    (y: 2) * 5 = ``,
    y: 12 = ``,
},

# Circular definition with internal definition
{
    6 - (x: (4 * y.h) - 3) = ``,
    y: {h: (1 - 44 / z)},
    4 * (z: x - 2) - 3 = ``,
},

