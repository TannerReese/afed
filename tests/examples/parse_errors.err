Parse Error in line 4, column 13
	{ x: 4, x: 5},
	            ^
Error: Redefinition of label 'x' in map

Parse Error in line 14, column 27
	{1, "zing zing", 4n  kd ( $ $, "this is": "correct"} = ``,
	                          ^
Error: Missing value

Parse Error in line 15, column 24
	{ "good", second: njd %% & 62 ] , third: "good"} = ``,
	                       ^
Error: Missing value

Parse Error in line 16, column 20
	["this", "next"  ( % ^&, "last"] = ``,
	                   ^
Error: Missing value

Parse Error in line 21, column 20
	    4 * (x: 3) = ``,
	                   ^
Error: Redefinition of label 'x' in map

Parse Error in line 23, column 15
	    y: 12 = ``,
	              ^
Error: Redefinition of label 'y' in map

6 Parse Errors encountered


line 8, column 26 Eval Error: Circular dependency
line 9, column 26 Eval Error: Circular dependency
line 10, column 26 Eval Error: Circular dependency
line 28, column 30 Eval Error: Circular dependency
line 30, column 26 Eval Error: Circular dependency
5 Eval Errors encountered
