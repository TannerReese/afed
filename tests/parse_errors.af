# Test parsing errors

# Redefinition error
{ x: 4, x: 5},

# Circular dependence error
{
    q: q, q = ``,
    a: b, b: c, c: a, a = ``,
    foo: {x: 5 * bar.y = ``},
    bar: {y: baz.z - 2 = ``},
    baz: {z: 7 / foo.x = ``},
},

# Extra content in map or array
{1, "zing zing", 4n  kd ( $ $, "this is": "correct"} = ``,
{ "good", second: njd %% & 62 ] , third: "good"} = ``,
["this", "next"  ( % ^&, "last"] = ``,

# Redefinition with internal definition
{
    x: 4 = ``,
    4 * (x: 3) = ``,
    (y: 2) * 5 = ``,
    y: 12 = ``,
},

# Circular definition with internal definition
{
    6 - (x: (4 * y.h) - 3) = ``,
    y: {h: (1 - 44 / z)},
    4 * (z: x - 2) - 3 = ``,
},


# Redefinition of map target
{
	x: 4,
	(hello: 3) + 1 = `4`,
	(: 5 + 3) - x = `4`,
	3 * x = `12`,
	x / hello ^ (: 2) = `Eval Error: Unresolved name "x"`,
	hello + x = `7`,
} = `8`,

