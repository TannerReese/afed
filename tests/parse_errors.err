Parse Error in line 4, column 13 of "parse_errors.af"
|   { x: 4, x: 5},
|               ^
Error: Redefinition of label 'x' in map

Parse Error in line 16, column 24 of "parse_errors.af"
|   {1, "zing zing", 4n  kd ( $ $, "this is": "correct"} = ``,
|                          ^
Error: Ill formed member

Parse Error in line 17, column 22 of "parse_errors.af"
|   { "good", second: njd %% & 62 ] , third: "good"} = ``,
|                        ^
Error: Ill formed member

Parse Error in line 18, column 16 of "parse_errors.af"
|   ["this", "next"  ( % ^&, "last"] = ``,
|                  ^
Error: Ill formed member

Parse Error in line 23, column 21 of "parse_errors.af"
|       4 * (x: 3) = ``,
|                       ^
Error: Redefinition of label 'x' in map

Parse Error in line 25, column 16 of "parse_errors.af"
|       y: 12 = ``,
|                  ^
Error: Redefinition of label 'y' in map

Parse Error in line 42, column 56 of "parse_errors.af"
|   	x / hello ^ (: 2) = `Eval Error: Unresolved name "x"`,
|                                                          ^
Error: Redefinition of map target

7 Parse Errors encountered


line 8, column 15 of "parse_errors.af" Eval Error: Circular dependency
line 9, column 27 of "parse_errors.af" Eval Error: Circular dependency
line 10, column 26 of "parse_errors.af" Eval Error: Circular dependency
line 11, column 26 of "parse_errors.af" Eval Error: Circular dependency
line 12, column 26 of "parse_errors.af" Eval Error: Circular dependency
line 30, column 30 of "parse_errors.af" Eval Error: Circular dependency
line 32, column 26 of "parse_errors.af" Eval Error: Circular dependency
line 42, column 22 of "parse_errors.af" Eval Error: Unresolved name "x"
8 Eval Errors encountered
