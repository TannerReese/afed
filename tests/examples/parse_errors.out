# Test parsing errors

# Redefinition error
{ x: 4, x: 5},

# Circular dependence error
{
    foo: {x: 5 * bar.y = `Eval Error: Circular dependency`},
    bar: {y: baz.z - 2 = `Eval Error: Circular dependency`},
    baz: {z: 7 / foo.x = `Eval Error: Circular dependency`},
},

# Extra content in map or array
{1, "zing zing", 4n  kd ( $ $, "this is": "correct"} = `{this is: "correct"}`,
{ "good", second: njd %% & 62 ] , third: "good"} = `{third: "good"}`,
["this", "next"  ( % ^&, "last"] = `["this", "last"]`,

# Redefinition with internal definition
{
    x: 4 = `4`,
    4 * (x: 3) = `12`,
    (y: 2) * 5 = `10`,
    y: 12 = `12`,
},

# Circular definition with internal definition
{
    6 - (x: (4 * y.h) - 3) = `Eval Error: Circular dependency`,
    y: {h: (1 - 44 / z)},
    4 * (z: x - 2) - 3 = `Eval Error: Circular dependency`,
},

